src/program.cpp

```diff
@@ -1521,6 +1552,26 @@ cl_int cvk_entry_point::init() {
         m_pod_buffer_size = round_up(m_pod_buffer_size, 4);
     }
 
+    // Take the size of image metadata into account for the pod buffer size
+    bool image_md_is_max = false;
+    for (auto md : m_image_channel_metadata) {
+        auto order_offset = md.second.first;
+        auto data_type_offset = md.second.second;
+        if (order_offset != IMAGE_METADATA_EMPTY &&
+            order_offset > m_pod_buffer_size) {
+            m_pod_buffer_size = order_offset;
+            image_md_is_max = true;
+        }
+        if (data_type_offset != IMAGE_METADATA_EMPTY &&
+            data_type_offset > m_pod_buffer_size) {
+            m_pod_buffer_size = data_type_offset;
+            image_md_is_max = true;
+        }
+    }
+    if (image_md_is_max) {
+        m_pod_buffer_size = round_up(m_pod_buffer_size + 4, 4);
```

Comment:
Why `+4`?

Reply:
sorry for that, just the size of the offset. changed it for `sizeof(uint32_t)`